ui/static/ui/js/manage_taxonomies.jsx

```diff
@@ -289,8 +289,7 @@ define('setup_manage_taxonomies', ['reactaddons', 'lodash', 'jquery', 'utils'],
         vocabulary: vocab
       };
       var vocabularies = this.state.vocabularies;
-      vocabularies.push(newVocab);
-      this.setState({vocabularies: vocabularies});
+      this.setState({vocabularies: vocabularies.concat([newVocab])});
```

Comment:
I guess I didn't realize this was mutating the state, since I'm calling set state anyway, how is this "forbidden"? (I'm just honestly curious why this is wrong)